ifrs17/Utils/Queries.ipynb

```diff
@@ -399,6 +400,66 @@
             "metadata": {},
             "execution_count": 0,
             "outputs": []
+        },
+        {
+            "cell_type": "markdown",
+            "source": [
+                "# Data Variables"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "public static async Task<T[]> LoadPartitionedDataAsync<T,P>(this IDataSource querySource, Guid partition)",
+                "\n    where T : IPartitioned ",
+                "\n    where P : IPartition",
+                "\n{",
+                "\n    await querySource.Partition.SetAsync<P>(partition);",
+                "\n    return (await querySource.Query<T>().ToArrayAsync());",
+                "\n}"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "public static async Task<T[]> RelaxedQueryPartitionedDataAsync<T,P>(this IWorkspace workspace, IDataSource dataSource, Guid targetPartition, Guid defaultPartition, string format)",
+                "\n    where T : IPartitioned ",
+                "\n    where P : IPartition",
+                "\n{",
+                "\n    var variables = await workspace.LoadPartitionedDataAsync<T,P>(targetPartition);",
+                "\n",
+                "\n    if(targetPartition != defaultPartition) {",
+                "\n        var isRelaxed = (format != ImportFormats.Cashflow && typeof(T).Name == nameof(IfrsVariable)) || ",
+                "\n                        (format == ImportFormats.Cashflow && typeof(T).Name == nameof(RawVariable));",
+                "\n        // For scenario re-calculation",
+                "\n        if(!variables.Any() && isRelaxed) variables = await dataSource.LoadPartitionedDataAsync<T,P>(targetPartition); ",
```

Comment:
I really don't like this way of writing, as the if statement looks the same, but it's not because of the change in variables.
Can you rewrite this?

extract isRelaxed in a separated if --> if false, you don't need anything else.

await dataSource.LoadPartitionedDataAsync<T,P>(targetPartition); you need this query anyway --> extract this in a variable outside the if (var targetVariables) --> then you if this --> variables = targetVariables.Any() ?  targetVariables : await dataSource.LoadPartitionedDataAsync<T,P>(defaultPartition);

something like that

Reply:
done